utils/generate-rubocop-rules/src/main/java/org/sonarsource/slang/externalreport/rubocop/RuboCopRuleDefinition.java

```diff
@@ -0,0 +1,138 @@
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.slang.externalreport.rubocop;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.text.StringEscapeUtils;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class RuboCopRuleDefinition {
+
+  private static final Path RULES_FILE = Paths.get("sonar-ruby-plugin", "src", "main", "resources",
+    "org", "sonar", "l10n", "ruby", "rules", "rubocop", "rules.json");
+
+  private static final Path RUBOCOP_HELP = Paths.get("utils", "generate-rubocop-rules",
+    "src", "main", "resources", "rubocop.yml");
+
+  public static void main(String[] args) throws IOException {
+    String rules = RuboCopRuleDefinitionGenerator.generateRuleDefinitionJson(resolve(RUBOCOP_HELP));
+    Files.write(resolve(RULES_FILE), rules.getBytes(UTF_8));
+  }
+
+  private static Path resolve(Path path) throws IOException {
+    Path projectPath = Paths.get(".").toRealPath();
+    while (!projectPath.resolve(path).toFile().exists()) {
+      projectPath = projectPath.getParent();
+    }
+    return projectPath.resolve(path);
+  }
+
+  static final class RuboCopRuleDefinitionGenerator {
+
+    private RuboCopRuleDefinitionGenerator() {
+      // Utility class
+    }
+
+    static String generateRuleDefinitionJson(Path ruboCopHelpPath) throws IOException {
+      RuboCopHelp help = new RuboCopHelp(ruboCopHelpPath);
+      help.rules.sort(Comparator.comparing(a -> a.key));
+      Gson gson = new GsonBuilder().disableHtmlEscaping().setPrettyPrinting().create();
+      return gson.toJson(help.rules) + "\n";
+    }
+  }
+
+  private static class ExternalRule {
+    String key;
+    String name;
+    String type;
+    String severity;
+    String description;
+    String url;
+  }
+
+  private static class RuboCopHelp {
```

Comment:
Why "help"?

Reply:
The name is related to the `rubocop --show-cops` command line output. I called it help, because it's the rubocop command line help. But in fact its a yaml structured list of rules. So I will change it to `RuboCopRulesYamlFile`